streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -204,4 +210,22 @@ public synchronized void addGlobalStore(final StoreBuilder<KeyValueStore> storeB
                        processorName,
                        stateUpdateSupplier);
     }
+
+    //TODO uncomment and implement when ready
+    public void buildAndOptimize() {
+//        if (topologyNeedsBuilding) {
+//            TopologyOptimizer topologyOptimizer = new TopologyOptimizerImpl();
+//            topologyOptimizer.optimize(topologyGraph, internalTopologyBuilder);
+//            topologyNeedsBuilding = false;
+//        }
+    }
+
+    // for testing
+    StreamsTopologyGraph getTopologyGraph() {
+        return topologyGraph;
+    }
+
+    public void addNode(StreamsGraphNode graphNode) {
```

Comment:
Why add this wrapper? Could we just call `topologyGraph.addNode(graphNode)` directly?

Reply:
ack see above